src/overview.md

```diff
@@ -0,0 +1,368 @@
+
+[`TyCtxt`]: https://doc.rust-lang.org/nightly/nightly-rustc/rustc/ty/struct.TyCtxt.html
+
+### `ty::Ty`
+
+Types are really important in Rust, and they form the core of a lot of compiler
+analyses. The main type (in the compiler) that represents types (in the user's
+program) is [`rustc::ty::Ty`][ty]. This is so important that we have a whole chapter
+on [`ty::Ty`][ty], but for now, we just want to mention that it exists and is the way
+`rustc` represents types!
+
+Oh, and also the `rustc::ty` module defines the `TyCtxt` struct we mentioned before.
+
+[ty]: https://doc.rust-lang.org/nightly/nightly-rustc/rustc/ty/type.Ty.html
+
+### Parallelism
+
+Compiler performance is a problem that we would like to improve on
+(and are always working on). One aspect of that is parallelizing
+`rustc` itself.
+
+Currently, there is only one part of rustc that is already parallel: codegen.
+During monomorphization, the compiler will split up all the code to be
+generated into smaller chunks called _codegen units_. These are then generated
+by independent instances of LLVM. Since they are independent, we can run them
+in parallel. At the end, the linker is run to combine all the codegen units
+together into one binary.
+
+However, the rest of the compiler is still not yet parallel. There have been
+lots of efforts spent on this, but it is generally a hard problem. The current
+approach is (**TODO: verify**) to turn `RefCell`s into `Mutex`s -- that is, we
+switch to thread-safe internal mutability. However, there are ongoing
+challenges with lock contention, maintaining query-system invariants under
+concurrency, and the complexity of the code base. One can try out the current
+work by enabling parallel compilation in `config.toml`. It's still early days,
+but there are already some promising performance improvements.
+
+### Bootstrapping
+
+`rustc` itself is written in Rust. So how do we compile the compiler? We use an
+older compiler to compile the newer compiler. This is called _bootstrapping_.
+
+Bootstrapping has a lot of interesting implications. For example, it means that one
+of the major users of Rust is Rust, so we are constantly testing our own
+software ("eating our own dogfood"). Also, it means building the compiler can
+take a long time because one must first build the compiler and then use that to
+build the new compiler (sometimes you can get away without the full 2-stage
+build, but for release artifacts you need the 2-stage build).
+
+Bootstrapping also has implications for when features are usable in the
+compiler itself. The build system uses the current beta compiler to build the
+stage-1 bootstrapping compiler. This means that the compiler source code can't
+use some features until they reach beta (because otherwise the beta compiler
+doesn't support them). On the other hand, for compiler intrinsics and internal
+features, we may be able to use them immediately because the stage-1
+bootstrapping compiler will support them.
```

Comment:
Also unsure if this has the right context to be understood.

Reply:
added to followup issue